indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -64,26 +65,22 @@ protected void map(
   ) throws IOException, InterruptedException
   {
     try {
-      final InputRow inputRow;
-      try {
-        inputRow = parseInputRow(value, parser);
-      }
-      catch (Exception e) {
-        if (config.isIgnoreInvalidRows()) {
-          log.debug(e, "Ignoring invalid row [%s] due to parsing error", value.toString());
-          context.getCounter(HadoopDruidIndexerConfig.IndexJobCounters.INVALID_ROW_COUNTER).increment(1);
-          return; // we're ignoring this invalid row
-        } else {
-          throw e;
-        }
-      }
-
+      final InputRow inputRow = parseInputRow(value, parser);
       if (!granularitySpec.bucketIntervals().isPresent()
           || granularitySpec.bucketInterval(new DateTime(inputRow.getTimestampFromEpoch()))
                             .isPresent()) {
         innerMap(inputRow, value, context);
       }
     }
+    catch (ParseException e) {
+      if (config.isIgnoreInvalidRows()) {
```

Comment:
can we flip this around for readability's sake and put the `isIgnoreInvalidRows` outside the `if`?